checker/checker.go

```diff
@@ -323,46 +323,46 @@ func (c *checker) checkBlockStmt(scope *parser.Scope, typ parser.ObjType, block
 		}
 
 		lookupByType := builtin.Lookup.ByType[typ]
-		if !foundSource {
-			// If the function is not a builtin, retrieve it from the scope and then
-			// type check it.
-			_, ok := lookupByType.Func[name]
-			if !ok {
-				obj := scope.Lookup(name)
-				if obj == nil {
-					return ErrFirstSource{call}
-				}
+		// If the function is not a builtin, retrieve it from the scope and then
+		// type check it.
+		_, ok := lookupByType.Func[name]
+		if !ok {
```

Comment:
I think we can make this implementation more clean. We should do branch off here for:
```go
		_, ok := lookupByType.Func[name]
		if ok {
			// Handle builtins
			// lookupByType.Func[name].IsSource ...
		} else {
			// Handle user-defined
		}

		// Setting some shared variable between the two branches, we
		// abstract away builtin vs. user-defined nad check ErrOnlyFirstSource here.
```